### PR TITLE
pyCUSFM/plotters.py

### PR DESCRIPTION
**Pull request by joelbenjohn**
Added crossect, dispshap and thecurve3

### CODE REVIEW

File: example_1.py [hunk 1]
```diff
@@ -1,117 +0,0 @@
-# This example presents a very simple Cee section,
-# solved for pure compression,
-# in the Imperial unit system
```
Comment (smith120bh) [inline example_1.py:3]: Why did you delete the examples? I think they're a very important part of users understanding how to use pyCUFSM. They certainly don't do any harm. I also plan to use them in automated validation testing of pyCUFSM. Unless you can give a really good reason for deleting them, can you please restore them?

File: example_2.py [hunk 2]
```diff
@@ -1,116 +0,0 @@
-# This example presents a very simple Zed section,
-# solved for pure bending about the X-axis,
-# in the Metric unit system
-
```
Comment (smith120bh) [inline example_2.py:4]: See comment on example 1 above

File: Jupyter_Notebooks/PreProcess.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
+import ipywidgets as widgets
+import numpy as np
+import sys
+import scipy.io
+sys.path.insert(1, 'C:/Users/joelb/pyCUFSM')
```
Comment (smith120bh) [inline Jupyter_Notebooks/PreProcess.py:5]: I'm not super familiar with Jupyter Notebook's programming specifically, but there definitely shouldn't be paths pointing to your personal local computer committed onto GitHub!

File: Jupyter_Notebooks/PreProcess.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+#     children = [page, page]
+#     tab = widgets.Tab()
+#     tab.children = children
+#     tab.set_title(0, 'Main Preprocessor')
+#     tab.set_title(1, 'Lengths and Boundary Conditions')
+#     return tab
+
+class Preprocess:
+
+    def wprops(self, props, m):
```
Comment (smith120bh) [inline Jupyter_Notebooks/PreProcess.py:33]: Can you add comments at least at the beginning of each function saying what the function does? It's very difficult to follow the code in here.

File: pycufsm/fsm.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import numpy as np
 import pycufsm.analysis
 import pycufsm.cfsm
-
+from scipy.sparse.linalg import eigs
```
Comment (smith120bh) [inline pycufsm/fsm.py:6]: This import is unused. Please comment it out

File: pycufsm/fsm.py [hunk 6]
```diff
@@ -333,45 +334,43 @@ def strip(
...
-        #     # eigs
-        # else:
-        #     eig_sparse = False
-        #     # eig
+        if rcond_num >= np.spacing(1.0) / 2:
+            eig_sparse = True
+            # eigs
+        else:
+            eig_sparse = False
+            # eig
```
Comment (smith120bh) [inline pycufsm/fsm.py:352]: The eig_sparse variable still isn't used for anything. Leave this commented out.

File: pycufsm/fsm.py [hunk 7]
```diff
@@ -413,4 +412,4 @@ def strip(
         # shapes(:,i,1:min([n_modes,num_pos_modes]))=modes
         shapes.append(modes_full)
 
-    return signature.flatten(), curve, shapes
+    return signature.flatten(), curve, shapes
```
Comment (smith120bh) [inline pycufsm/fsm.py:415]: Re-add a newline at the end of the file here. I'm surprised YAPF didn't give you a warning about this.

File: pycufsm/preprocess.py [hunk 8]
```diff
@@ -364,22 +364,91 @@ def template_out_to_in(sect):
...
-        l_1 = sect['l'] - (rad + thick/2)*np.tan(np.pi/4)
-    if sect['l'] == 0:
+        l_1 = sect['d_1'] - (rad + thick/2)*np.tan(np.pi/4)
+    if sect['d'] == 0:
         b_2 = b_2 - rad - thick/2
         l_2 = 0
     else:
         b_2 = b_2 - rad - thick/2 - (rad + thick/2)*np.tan(np.pi/4)
-        l_2 = sect['l'] - (rad + thick/2)*np.tan(np.pi/4)
+        l_2 = sect['d_2'] - (rad + thick/2)*np.tan(np.pi/4)
```
Comment (smith120bh) [inline pycufsm/preprocess.py:380]: a) I know CUFSM labeled lip length as `d` - but most manufacturer catalogs call it `l` and the internal variables are all `l`. I strongly suggest we use `l` and not `d`.
b) If you're assuming the lip length is defined as `l_1` and `l_2`, then the conditionals need to be on `l_1` and `l_2` - not just `l`.

**Review by smith120bh — CHANGES_REQUESTED** [review]
Please see specific comments in files. The most significant points are that I ask you to please not delete the example files, and please comment your new code - _at least_ with a comment at the beginning of each function explaining what the function does.

File: pycufsm/fsm.py (hunk 6 above)
Comment (joelbenjohn) [reply pycufsm/fsm.py:352]: Will do

File: pycufsm/preprocess.py (hunk 8 above)
Comment (joelbenjohn) [reply pycufsm/preprocess.py:380]: I couldn't find any place where the template_out_to_in and template_calc functions were being used..
I made the changes while trying to make a Jupyter Notebook which could load SFIA sections to analyze.
Will change it back to 'l'

File: example_1.py (hunk 1 above)
Comment (joelbenjohn) [reply example_1.py:3]: Didn't mean to delete..didn't want to commit the changes I made into examples 1 and 2 into the main repo so didn't commit into mine and ended up deleting from mine..
Will add it back

File: Jupyter_Notebooks/PreProcess.py (hunk 3 above)
Comment (joelbenjohn) [reply Jupyter_Notebooks/PreProcess.py:5]: Yes..sorry the Jupyter Notebook files weren't ready to commit yet(added them to my repo to test the notebooks on Colab)..And this particular script isn't gonna be in the final version..
Thought I'd pull request on some of the changes(plotters.py) and once they were reviewed would work on adding the rest of the files

File: Jupyter_Notebooks/PreProcess.py (hunk 4 above)
Comment (joelbenjohn) [reply Jupyter_Notebooks/PreProcess.py:33]: Yes this script isn't gonna be in the final version..didn't mean to add this to the main repo

File: pycufsm/preprocess.py (hunk 8 above)
Comment (joelbenjohn) [reply pycufsm/preprocess.py:380]: image

I used ipywidgets package to make this template..Since you already wrote the template_out_to_in and template_calc in the preprocess, getting this to work was a lot easier..sorry about changing 'l' to 'd'..I was trying to understand the code comparing it to the matlab version so changed it to 'd'

File: example_1.py (hunk 1 above)
Comment (smith120bh) [reply example_1.py:3]: It may be easier to create a new branch on _your_ fork, and pull request that _branch_ into here. Right now, you're pull requesting your fork's `master` into here, which may make it a bit more difficult to exclude changes you don't want to commit. (This doesn't matter to me - just may be easier for you to work with. I have a couple upstream projects I contribute to for which I do this)

File: Jupyter_Notebooks/PreProcess.py (hunk 3 above)
Comment (smith120bh) [reply Jupyter_Notebooks/PreProcess.py:5]: Yes, committing in small chunks makes it faster and easier for me to review too :-)  You could either edit your branch here, or cancel this pull request and create a new one of a more limited branch.

File: pycufsm/preprocess.py (hunk 8 above)
Comment (smith120bh) [reply pycufsm/preprocess.py:380]: No worries! Experimenting is needed to get things to work properly. The screenshot there looks nice! But yeah, since I was the one porting the code initially, I took a couple liberties to clean up things that didn't make sense - like always indicating the minor axis with `z` (it was randomly `y` in a couple places), and not using `d` for the lip length. :-)
